TeaDiary/Scripts/App/TeaListDisplayer.js

```diff
@@ -32,30 +44,71 @@
     }
 
     function buildHeader(teas) {
-        if (teas != null && teas.length > 0) {
+        if (!isEmpty(teas)) {
             return $("<h1>").html(TEA_LIST_TITLE);
         } else {
             return $("<h1>").html(NO_TEAS_TITLE);
         }
     }
 
-    function buildElement(tea) {
-        var teaString = $('<div>')
-            .html($("<a>")
-                .attr('href', TEA_DETAILS_PAGE_LINK + tea.Id)
-                .attr("id", TEA_ELEMENT_ID_PREFIX + tea.Id)
-                .text(tea.Name + ' (' + tea.Type + ')')
-            );
-        return teaString;
+    function buildElement(tea, counter) {
+        var teaElement = $('<tr>')
+            .attr("id", TEA_ELEMENT.PREFIX + TEA_ELEMENT.ID + tea.Id);
+        var teaNumber = $("<td>")
+            .text(counter);
+        var teaName = $("<td>")
+            .text(tea.Name);
+        var teaType = $("<td>")
+            .text(tea.Type);
+        var teaLink = $("<td>")
+            .addClass("link text-info")
+            .attr("id", TEA_ELEMENT.PREFIX + TEA_ELEMENT.LINK + tea.Id)
+            .attr('href', TEA_DETAILS_PAGE_LINK + tea.Id)
+            .text(MOAR);
+
+        appendElement(teaNumber, teaElement);
+        appendElement(teaName, teaElement);
+        appendElement(teaType, teaElement);
+        appendElement(teaLink, teaElement);
+        return teaElement;
     }
 
-    function appendElement(text) {
-        self.container.append(text);
+    function appendElement(text, container) {
+        var config = (container == null) ? teaListContainer : container;
+        $(config).append(text);
     }
 
     function displayTea(tea) {
-        var teaLine = buildElement(tea);
+        var teaLine = buildElement(tea, entityCounter);
+        entityCounter++;
         appendElement(teaLine);
     }
 
+    function displayHeader(teas) {
```

Comment:
Since you now have a table... Maybe the `h1` setup is on a different level than table headers? You can name it `title` and separate it from `headers`